app/routes/cdo/delete/dog.js

```diff
@@ -23,18 +24,18 @@ const dogAndOwnerConfirmation = {
       return h.redirect(`${routes.viewDogDetails.get}/${pk}${extractBackNavParam(request)}`).takeover()
     }
 
-    const backLink = `${routes.deleteDog.get}/${pk}?src=${extractBackNavParam(request)}`
+    const backNav = getBackLinkToSamePage(request)
```

Comment:
Are you losing the src param here? I guess it's not a problem since you'll have lost it when navigating from the admin index link anyway